encrypt/encrypt.go

```diff
@@ -58,109 +43,45 @@ import (
 // https://github.com/hashicorp/vault/blob/master/command/pgp_test.go
 
 // Decrypt a file with a provided key.
-func Decrypt(filename string, pubkey string, privkey string) {
-	decryptFile(pubkey, privkey, filename)
+func Decrypt(filename string, opts options.Options) {
+	decryptFile(filename, opts)
 }
 
 // Encrypt a file
-func Encrypt(filename string, pubkey string) {
-	encryptFile(pubkey, filename)
+func Encrypt(filename string, opts options.Options) {
+    log.Debug(opts)
+	encryptFile(filename, opts)
 }
 
-// This was an older deprecated function.
-func encrypt2(filename string, pubkey string) {
-	key := getKey(pubkey)
-	log.Debugf("\tPublic key: %v", key)
+func getKey(opts options.Options) *armor.Block {
+    var in io.Reader
+    var err error
 
-	// Read in public key
-	recipient, err := readEntity(key)
-	if err != nil {
-		fmt.Println(err)
-		return
-	}
+    // if provided SSM Pub Key, then fetch from SSM
+    if opts.SSMPubKey != "" {
+        in = strings.NewReader(aws_helpers.GetParameterValue(opts.SSMPubKey, opts))
 
-	f, err := os.Open(filename)
-	if err != nil {
-		log.Error(err)
-		return
-	}
```

Comment:
Do we see a scenario where the public key will be accessed from SSM before encrypting the file? How will the access be setup? Will it make things easier, if the only option to encrypt is to use physical key file stored on the server? I think that is reasonable as setting up the public key will be one time setup step

Reply:
The current implementation of this only allows for SSM arguments for the decrypt command. 

Even though both share and decrypt commands rely on the encrypt package, the share command does not allow for SSM arguments, so the current design requires a present key on the server. Additionally, we could provision the IAM role we give the tenant to not be able to read SSM parameters as an additional step. 

What are your thoughts?

Reply:
Let us keep it that way - we do not want to read the public key from SSM. This will keep the options and setup simpler